src/app/ReadClient.cpp

```diff
@@ -520,6 +520,15 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
 CHIP_ERROR ReadClient::RefreshLivenessCheckTimer()
 {
     CancelLivenessCheckTimer();
+    uint64_t timeoutMsec = mMaxIntervalCeilingSeconds;
+    uint64_t marginSec   = 0;
+    if (mpExchangeCtx != nullptr && mpExchangeCtx->IsUDPTransport())
+    {
+        marginSec = (CHIP_CONFIG_RMP_DEFAULT_MAX_RETRANS + 1) * mpExchangeCtx->GetIdleRetransmitTimeoutTick();
+    }
+
+    timeoutMsec = timeoutMsec + marginSec * 1000 /* MILLISECOND_PER_SECOND 1000*/;
```

Comment:
I suspect using the System::Clock types here would avoid the need for magic constants and get us stronger typing and mixup-avoidance in the process.

Reply:
Done, thanks